server/lib/nl/fulfillment/triple.py

```diff
@@ -44,20 +45,34 @@ def _entity_name(entity: Entity) -> str:
 
 # Gets a title string for a list of entities
 def _get_entity_string(entities: List[Entity]) -> str:
-  if len(entities) == 1:
-    return _entity_name(entities[0])
+  # Get the list of unique entity names
+  seen_entity_names = set()
+  unique_entity_names = []
+  for e in entities:
+    entity_name = _entity_name(e)
+    entity_name_lowercase = entity_name.lower()
+    if not entity_name_lowercase in seen_entity_names:
+      unique_entity_names.append(entity_name)
+    seen_entity_names.add(entity_name_lowercase)
```

Comment:
super nit: add to set when non-existent

Reply:
sorry, what do you mean add to set when non-existent?

Reply:
Do we need to insert seen_entity_names only under the if condition?

Reply:
ohh, yes, updated